src/routers/StackRouter.js

```diff
@@ -102,23 +109,38 @@ export default (routeConfigs, stackConfig = {}) => {
       // Set up the initial state if needed
       if (!state) {
         let route = {};
-        if (
-          behavesLikePushAction(action) &&
-          childRouters[action.routeName] !== undefined
-        ) {
+        const childRouter = childRouters[action.routeName];
+        if (behavesLikePushAction(action) && childRouter !== undefined) {
```

Comment:
I had to rebase some changes that caused a conflict here, double check this is still the right behavior plz.